presto-native-execution/presto_cpp/main/SystemConnector.cpp

```diff
@@ -0,0 +1,174 @@
+    temp = fmt::format("{}", task->id.queryId());
+    taskTableResult->childAt(4)->asFlatVector<StringView>()->set(i, StringView(temp));
+    temp = fmt::format("{}", taskInfo.taskStatus.state);
+    taskTableResult->childAt(5)->asFlatVector<StringView>()->set(i, StringView(temp));
+    taskTableResult->childAt(6)->asFlatVector<int64_t>()->set(i, taskInfo.stats.totalDrivers);
+    taskTableResult->childAt(7)->asFlatVector<int64_t>()->set(i, taskInfo.stats.queuedDrivers);
+    taskTableResult->childAt(8)->asFlatVector<int64_t>()->set(i, taskInfo.stats.runningDrivers);
+    taskTableResult->childAt(9)->asFlatVector<int64_t>()->set(i, taskInfo.stats.completedDrivers);
+    taskTableResult->childAt(10)->asFlatVector<int64_t>()->set(i, toMillis(taskInfo.stats.totalScheduledTimeInNanos));
+    taskTableResult->childAt(11)->asFlatVector<int64_t>()->set(i, toMillis(taskInfo.stats.totalCpuTimeInNanos));
+    taskTableResult->childAt(12)->asFlatVector<int64_t>()->set(i, toMillis(taskInfo.stats.totalBlockedTimeInNanos));
+    taskTableResult->childAt(13)->asFlatVector<int64_t>()->set(i, taskInfo.stats.rawInputDataSizeInBytes);
+    taskTableResult->childAt(14)->asFlatVector<int64_t>()->set(i, taskInfo.stats.rawInputPositions);
+    taskTableResult->childAt(15)->asFlatVector<int64_t>()->set(i, taskInfo.stats.processedInputDataSizeInBytes);
+    taskTableResult->childAt(16)->asFlatVector<int64_t>()->set(i, taskInfo.stats.processedInputPositions);
+    taskTableResult->childAt(17)->asFlatVector<int64_t>()->set(i, taskInfo.stats.outputDataSizeInBytes);
+    taskTableResult->childAt(18)->asFlatVector<int64_t>()->set(i, taskInfo.stats.outputPositions);
+    taskTableResult->childAt(19)->asFlatVector<int64_t>()->set(i, taskInfo.stats.physicalWrittenDataSizeInBytes);
+    // Setting empty TIMESTAMP for below fields as string -> Timestamp is not implemented in Velox.
+    //taskTableResult->childAt(20)->asFlatVector<Timestamp>()->set(i, Timestamp(taskInfo.stats.createTime));
+    taskTableResult->childAt(20)->asFlatVector<Timestamp>()->set(i, Timestamp());
+    //taskTableResult->childAt(21)->asFlatVector<Timestamp>()->set(i, Timestamp(taskInfo.stats.firstStartTime));
+    taskTableResult->childAt(21)->asFlatVector<Timestamp>()->set(i, Timestamp());
+    //taskTableResult->childAt(22)->asFlatVector<Timestamp>()->set(i, Timestamp(taskInfo.lastHeartbeat));
+    taskTableResult->childAt(22)->asFlatVector<Timestamp>()->set(i, Timestamp());
+    //taskTableResult->childAt(23)->asFlatVector<Timestamp>()->set(i, Timestamp(taskInfo.stats.endTime));
+    taskTableResult->childAt(23)->asFlatVector<Timestamp>()->set(i, Timestamp());
+
+    i++;
+  }
+}
+
+}
+
+std::optional<RowVectorPtr> SystemDataSource::next(
+    uint64_t size,
+    velox::ContinueFuture& /*future*/) {
+  if (!currentSplit_) {
+    return nullptr;
+  }
+
+  TaskMap taskMap = taskManager_->tasks();
+  auto numRows = taskMap.size();
+  getTaskResults(taskMap, taskTableResult_);
```

Comment:
We can avoid `taskTableResult_` if we pass the desired columns (`outputColumnMappings_`) and `result` vector to `getTaskResults`.

Reply:
getTaskResults can be a private method and it can return RowVector directly.

Reply:
@aditi-pandit What do you think?

Reply:
That's a good idea. Have changed the code.